[test optimization] Add known limitation and workaround for vitest (#36345)
DataDog/documentation: content/en/tests/setup/javascript.md

### Vitest's browser mode
Vitest's [browser mode][17] is not supported.

> ### Vitest's test duration overhead

Reviewer: ```suggestion
### Vitest's test duration overhead

```

> By default, Vitest uses `isolate: true` (see [Vitest configuration][21]), so each test file runs in its own fork or thread. Vitest is ESM-first and relies on [import-in-the-middle][20] for instrumentation. That library pays a setup cost every time a suite starts; with isolation, that cost is repeated for every file. The effect is largest when you have many small, fast suites, because setup time can dominate wall-clock time.

Reviewer: ```suggestion
By default, Vitest's [`isolate`][21] option is `true`, so each test file runs in its own fork or thread. Vitest is ESM-first and relies on [import-in-the-middle][20] for instrumentation, which incurs a setup cost every time a suite starts. With isolation, that setup cost is repeated for every file. The effect is largest when you have many small, fast suites, because setup time can dominate wall-clock time.
```

> To lower overhead, set `isolate: false` in your `vitest.config` file, or pass `--no-isolate` to the test command.

Reviewer: Minor: I believe `vitest.config` isn't a real filename — the actual file is `vitest.config.js`/`.ts`/`.mjs`. Suggest dropping the code styling and saying "your Vitest config file," or being explicit with an example like `vitest.config.ts`.

```suggestion
To lower overhead, set `isolate: false` in your Vitest config file, or pass `--no-isolate` to the test command.
```
